src/shared/reportIssue.ts

```diff
@@ -0,0 +1,142 @@
+    dotnetResolver: IHostExecutableResolver,
+    monoResolver?: IHostExecutableResolver) {
+    // Get info for the dotnet that the language server executable is run on, not the dotnet the language server will execute user code on.
+    let fullDotnetInfo: string | undefined;
+    try {
+        const info = await dotnetResolver.getHostExecutableInfo(options);
+        const dotnetInfo = await getDotnetInfo([ dirname(info.path) ]);
+        fullDotnetInfo = dotnetInfo.FullInfo;
+    } catch (error) {
+        const message = error instanceof Error ? error.message : `${error}`;
+        fullDotnetInfo = message;
+    }
+
+    let monoInfo = "";
+    if (shouldIncludeMonoInfo && monoResolver) {
+        monoInfo = await getMonoIfPlatformValid(options, monoResolver);
+    }
+    
+    let extensions = getInstalledExtensions(vscode);
+
+    const useOmnisharp = options.commonOptions.useOmnisharpServer;
+    const logInfo = getLogInfo(useOmnisharp);
+
+    const body = `## Issue Description ##
+## Steps to Reproduce ##
+
+## Expected Behavior ##
+
+## Actual Behavior ##
+
+## Logs ##
+
+${logInfo}
+
+## Environment information ##
+
+**VSCode version**: ${vscode.version}
+**C# Extension**: ${csharpExtVersion}
+**Using OmniSharp**: ${useOmnisharp}
+
+${monoInfo}
+<details><summary>Dotnet Information</summary>
+${fullDotnetInfo}</details>
+<details><summary>Visual Studio Code Extensions</summary>
+${generateExtensionTable(extensions)}
+</details>
+`;
+
+    const queryStringPrefix: string = "?";
+    const issueDefault = "Please paste the output from your clipboard";
+    const fullUrl = `${issuesUrl}${queryStringPrefix}body=${issueDefault}`;
+    await vscode.env.clipboard.writeText(body);
+    vscode.env.openExternal(vscode.Uri.parse(fullUrl));
+}
+
+function sortExtensions(a: Extension<any>, b: Extension<any>): number {
+
+    if (a.packageJSON.name.toLowerCase() < b.packageJSON.name.toLowerCase()) {
+        return -1;
+    }
+    if (a.packageJSON.name.toLowerCase() > b.packageJSON.name.toLowerCase()) {
+        return 1;
+    }
+    return 0;
+}
+
+function generateExtensionTable(extensions: Extension<any>[]) {
+    if (extensions.length <= 0) {
+        return "none";
+    }
+
+    const tableHeader = `|Extension|Author|Version|Folder Name|\n|---|---|---|---|`;
+    const table = extensions.map((e) => `|${e.packageJSON.name}|${e.packageJSON.publisher}|${e.packageJSON.version}|${basename(e.extensionPath)}|`).join("\n");
+
+    const extensionTable = `
+${tableHeader}\n${table};
+`;
+
+    return extensionTable;
+}
+
+async function getMonoIfPlatformValid(options: Options, monoResolver: IHostExecutableResolver): Promise<string> {
+    let monoVersion = "Unknown Mono version";
+    try {
+        const monoInfo = await monoResolver.getHostExecutableInfo(options);
+        monoVersion = `OmniSharp using mono: ${monoInfo.version}`;
+    }
+    catch (error) {
+        monoVersion = `There is a problem with running OmniSharp on mono: ${error}`;
```

Comment:
The other place we're catching exceptions we have a different approach for converting it to a string (the instanceof check). Should we make them the same?

Reply:
Potentially - again I didn't modify this either (even though git says I did).  I'll fix it up though.